### PR TITLE
Fixed small bug in performance_metrics/forecasting/_functions.py

### PR DESCRIPTION
**Pull request by krumeto**
#### Reference Issues/PRs
There is no currently open issue or PR on this bug
#### What does this implement/fix? Explain your changes.
In the mase_loss function, there is a check if the training set is prior to test set. In the latest sktime version, the check was comparing y_train.index.max() >= y_test.min() , rather than y_train.index.max() >= y_test.index.min(). In addition, the ValueError raised a comment which was referring to y_pred, while the check performed was against y_test. Both were straightforward fixes.

#### Does your contribution introduce a new dependency? If yes, which one?

No new dependency.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Review by riteshahuja13 — DISMISSED** [review]
This is causing the function to break in my use case as well. #426 
With these changes it works well.

**Comment by mloning** [comment]
Thanks for the fix @krumeto and the feedback @riteshahuja13! Synced with master and will merge once all tests pass!
